cmd/api/controller/controller.go

```diff
@@ -17,18 +42,39 @@ func NewPessoaController(service pessoa.Service) *PessoaController {
 }
 
 func (c *PessoaController) Create(ctx *fiber.Ctx) error {
-	var pessoa *domain.Pessoa
+	pessoareq := PessoaRequest{}
 
-	if err := ctx.BodyParser(&pessoa); err != nil {
-		return err
+	validate := validator.New()
+
+	if err := ctx.BodyParser(&pessoareq); err != nil {
+		return fiber.NewError(http.StatusBadRequest, fmt.Sprintf("invalid json: %s", err.Error()))
+	}
+
+	if err := validate.Struct(&pessoareq); err != nil {
+		return fiber.NewError(http.StatusBadRequest, err.Error())
 	}
 
-	err := c.service.CreatePessoa(pessoa)
+	newPessoa := domain.Pessoa{
+		CpfCnpj:    pessoareq.CpfCnpj,
+		Name:       pessoareq.Name,
+		Nascimento: pessoareq.Nascimento,
+		Seguros:    pessoareq.Seguros,
+	}
+
+	pessoaSucess, err := c.service.CreatePessoa(&newPessoa)
 	if err != nil {
 		return err
 	}
 
-	return ctx.Status(fiber.StatusCreated).JSON(pessoa)
+	NewResponsePessoa := PessoaResponse{
+		ID:         pessoaSucess.ID,
+		CpfCnpj:    pessoaSucess.CpfCnpj,
+		Name:       pessoaSucess.Name,
+		Nascimento: pessoaSucess.Nascimento,
+		Seguros:    pessoaSucess.Seguros,
+	}
+
+	return ctx.Status(fiber.StatusCreated).JSON(NewResponsePessoa)
```

Comment:
Aqui eu tbm tenho um jeito "minimalista" de criar um DTO, como eu na verdade só quero uma coisa que gere um JSON eu prefiro não usar um struct: Eu uso um mapa mesmo, pq a sintaxe é mais sucinta, aí eu costumo fazer assim:

```suggestion
	return ctx.Status(fiber.StatusCreated).JSON(map[string]any{
		"id": pessoaSuccess.ID,
		"nome": pessoaSuccess.Name,
		"cpfcnpj": pessoaSuccess.CpfCnpj
		"nascimento": pessoaSuccess.Nascimento,
		"seguros": pessoaSuccess.Seguros,
	})
```

Fazendo isso eu nem preciso declarar o struct, o que já economiza um nome na cabeça do leitor, e eu tbm não gosto de salvar esse mapa numa variável intermediária para economizar mais um nome.

Não tem muita vantagem usar um struct aqui pq é um uso único então usar um mapa acaba tendo exatamente o mesmo efeito e sendo mais compacto.

Reply:
Ah eu tbm não costumo fazer validação de valores de resposta, já que se eu estiver retornando algo errado é erro que eu posso pegar nos testes.